sklearn/preprocessing/_polynomial.py

```diff
@@ -27,6 +32,68 @@
 ]
 
 
+def _create_expansion(X, interaction_only, deg, n_features, cumulative_size=0):
+    """Helper function for creating and appending sparse expansion matrices"""
+
+    total_nnz = _calc_total_nnz(X.indptr, interaction_only, deg)
+    expanded_col = _calc_expanded_nnz(n_features, interaction_only, deg)
+
+    if expanded_col == 0:
+        return None
+    assert expanded_col > 0
```

Comment:
Can we please avoid `assert` statements in productive code, e.g. everywhere except tests?
Even some linters/code helpers complain about it, see https://beta.ruff.rs/docs/rules/assert/.

Reply:
Good catch. I went ahead and removed it entirely, rather than replacing with an exception, since we know that `_calc_expanded_nnz(n_features, ...)` is non-negative for `n_features>=0`